### PR TITLE
Port CI tests to GitHub Actions

### LINKED ISSUE #301: Migrating from Travis CI to GitHub Actions?
Since travis-ci.org is shutting down, it probably makes sense to move this open project to another CI platform. GitHub actions seems like a low-friction environment.

Is this something the team would be interested in? If so, I would be willing to lead the effort to get a CI pipeline working again.

### PR DESCRIPTION
**Pull request by damienstanton**
This is a naive port of the current test suite over to GitHub Actions (GH Actions), and completes in 7-10 minutes. It is just a boilerplate runner that then executes the `test-all` script. Note that this GH Action does not replicate documentation generation and push. If that's a desired feature, please let me know and I'll update the PR to include it.

Fixes #301.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Build Status
+
+on:
+  push:
+    branches: [ main ]
```
Comment (dskkato) [inline .github/workflows/main.yml:5]: The default branch for this repository is `master`.

**Comment by damienstanton** [comment]
Hey @adamcrume, per @dskkato's comment w.r.t. the primary branch - are the maintainers okay with switching to `main`, or do other systems rely on that branch being called `master`?

**Comment by adamcrume** [comment]
We can't drop the `master` name entirely because it's referred to in a number of places.  It's mentioned in `src/lib.rs`, `CONTRIBUTING.md`, and `README.md` for example.  Those are easy to change at HEAD, but there are also generated artifacts (e.g. https://docs.rs/tensorflow/0.14.0/tensorflow) which link to files using the `master` branch name and can't be changed as well as other links out in the wild, peoples' upstream branch settings in their git repos, etc.

That said, we could switch the main branch name to `main` and keep `master` for backwards compatibility (probably keeping it up to date with `main` for some suitable period), and I'd also need to announce the change on the mailing list.

**Comment by damienstanton** [comment]
@googlebot I consent.

**Comment by damienstanton** [comment]
@adamcrume thanks, I've updated the config to use the `master` branch per your comment. I am not sure what is going on with the CLA now, however. I am comitting with the same git config as I did previously.

**Comment by adamcrume** [comment]
The CLA tool says this PR contains commits by other authors.  Can you try rebasing your changes on top of the latest `master` branch?

**Comment by adamcrume** [comment]
Since the migration to GitHub Actions was addressed by https://github.com/tensorflow/rust/pull/327, I'm closing this PR.

**Pull request closed without merging**